### DISCUSSION TITLE
blog/2024/asymmetric-routing-with-sr-linux-in-evpn-vxlan-fabrics/

### DISCUSSION THREAD

**Discussion by giscus** [srl-labs/learn-srlinux#175] [comments]
# blog/2024/asymmetric-routing-with-sr-linux-in-evpn-vxlan-fabrics/

Open documentation for open Network OS

https://asymmetric-evpnvxlan.learn-srlinux.pages.dev/blog/2024/asymmetric-routing-with-sr-linux-in-evpn-vxlan-fabrics/

**Comment by hasanashik** [comment]
Why there is not ip-vrf configured for inter vlan routing? How two irb subinterface will talk each other without ip vrf?

**Reply by aninchat** [reply]
I'm not sure if I understand the question correctly, but IP VRFs are not a necessity for inter-VLAN routing or asymmetric routing in EVPN VXLAN fabrics.

**Comment by tonybourke** [comment]
I see mostly symmetric IRB these days, what would be a reason one might choose asymmetric over symmetric?

**Reply by aninchat-nokia** [reply]
I agree, I think symmetric IRB is generally the preferred option in most cases. For asymmetric, the reasons are not as compelling but they could be - smaller scale fabric with low ARP/MAC scale and thus, consistent BD/IRB configuration across all VTEPs (easier to operate and maintain when using orchestrators or automation tools like Ansible). Blog (and a nokia validated design for symmetric IRB) will come in a few months.

**Comment by amp109** [comment]
Shouldn't the IRB interface be associated with an IP VRF?

A:leaf1# show network-instance macvrf1 interfaces
========================================================================================================================================================================================
Net instance    : macvrf1
Interface       : irb0.10
Oper state      : down
Oper down reason: ip-vrf-association-missing

**Reply by aninchat-nokia** [reply]
Do you have the IRB interface mapped to either an IP VRF or default network-instance? Please paste the MAC VRF and default or IP VRF network-instance configuration.

**Reply by amp109** [reply]
Thank you for your reply.
I tried to create the configuration from the article. So I need to associate the IRB with an IP-VRF or the default instance, correct?

 info network-instance macvrf1
    network-instance macvrf1 {
        type mac-vrf
        admin-state enable
        interface ethernet-1/3.0 {
        }
        interface irb0.10 {
        }
        vxlan-interface vxlan1.1 {
        }
        protocols {
            bgp-evpn {
                bgp-instance 1 {
                    admin-state enable
                    vxlan-interface vxlan1.1
                    evi 10
                    ecmp 2
                }
            }
            bgp-vpn {
                bgp-instance 1 {
                    route-distinguisher {
                        rd 192.0.2.11:1
                    }
                    route-target {
                        export-rt target:10:10
                        import-rt target:10:10
                    }
                }
            }
        }
    }

info network-instance default
    network-instance default {
        interface ethernet-1/1.0 {
        }
        interface ethernet-1/2.0 {
        }
        interface system0.0 {
        }
        protocols {
            bgp {
                admin-state enable
                autonomous-system 65411
                router-id 192.0.2.11
                afi-safi evpn {
                    admin-state enable
                }
                afi-safi ipv4-unicast {
                    admin-state enable
                    multipath {
                        maximum-paths 2
                    }
                }
                group spine {
                    peer-as 65500
                    export-policy [
                        spine-export
                    ]
                    import-policy [
                        spine-import
                    ]
                    afi-safi evpn {
                        admin-state enable
                    }
                    afi-safi ipv4-unicast {
                        admin-state enable
                    }
                }
                neighbor 198.51.100.1 {
                    peer-group spine
                }
                neighbor 198.51.100.3 {
                    peer-group spine
                }
            }
        }
    }

**Reply by aninchat-nokia** [reply]
Yes, the IRB interface must belong to a network-instance - either the default network-instance or an IP VRF network-instance. For asymmetric, it does not matter much since it works even with default network-instance (unless you want to provide L3 isolation, in which case, place it in an IP VRF network-instance).
